### PR TITLE
Sidebar refactoring

### PR DESCRIPTION
**Pull request by antoine-atmire**
The sidebar of the search page is currently baked into the search page component. In order to replicate the same kind of sidebar on other pages a reusable page-with-sidebar component has been created based on the search page component. 

This new component just takes care of displaying the sidebar properly on all viewports. It does not provide default buttons to open or close the sidebar. Instead the parent component is expected to provide the content of the sidebar through an input. The main content of the page goes in the template outlet (inside the page-width-sidebar tags). 

```
<ds-page-with-sidebar [id]="'search-page'" [sidebarContent]="sidebarContent">
  ... The main content of the page goes here
</ds-page-with-sidebar>

<ng-template #sidebarContent>
   ... The content of the sidebar goes here
</ng-template>
```
The width of the sidebar can also be passed as an input. The search page component has been updated to use this new component as well.

Some components that make up the search filters have also been recreated for the search-agnostic component. (A collapsable filter box, a dropdown, etc) However the original search components have not been removed and are still used by the search page.

### CODE REVIEW

File: src/app/shared/sidebar/filter/sidebar-filter.component.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import { SidebarFilterService } from './sidebar-filter.service';
+import { slide } from '../../animations/slide';
+
+@Component({
+  selector: 'ds-sidebar-filter',
+  styleUrls: ['./sidebar-filter.component.scss'],
+  templateUrl: './sidebar-filter.component.html',
+  animations: [slide],
+})
+export class SidebarFilterComponent implements OnInit {
```
Comment (tdonohue) [inline src/app/shared/sidebar/filter/sidebar-filter.component.ts]: Please add TypeDocs or a comment to describe this Component.

File: src/app/shared/sidebar/page-with-sidebar.component.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import { pushInOut } from '../animations/push';
+import { map } from 'rxjs/operators';
+
+@Component({
+  selector: 'ds-page-with-sidebar',
+  styleUrls: ['./page-with-sidebar.component.scss'],
+  templateUrl: './page-with-sidebar.component.html',
+  animations: [pushInOut],
+})
+export class PageWithSidebarComponent implements OnInit {
```
Comment (tdonohue) [inline src/app/shared/sidebar/page-with-sidebar.component.ts]: Please add Typedocs or a comment to describe this Component

File: src/app/shared/sidebar/sidebar-dropdown.component.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+
+@Component({
+  selector: 'ds-sidebar-dropdown',
+  styleUrls: ['./sidebar-dropdown.component.scss'],
+  templateUrl: './sidebar-dropdown.component.html',
+})
+export class SidebarDropdownComponent {
```
Comment (tdonohue) [inline src/app/shared/sidebar/sidebar-dropdown.component.ts]: Please add Typedocs or a comment to describe this Component

File: src/app/shared/sidebar/sidebar.service.ts [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
-const sidebarCollapsedSelector = createSelector(sidebarStateSelector, (sidebar: SearchSidebarState) => sidebar.sidebarCollapsed);
+const sidebarStateSelector = (state: AppState) => state.sidebar;
+const sidebarCollapsedSelector = createSelector(sidebarStateSelector, (sidebar: SidebarState) => sidebar.sidebarCollapsed);
 
 /**
  * Service that performs all actions that have to do with the search sidebar
  */
 @Injectable()
-export class SearchSidebarService {
+export class SidebarService {
```
Comment (tdonohue) [inline src/app/shared/sidebar/sidebar.service.ts]: We should update/correct the comment that describes this Service (just above this line), as it no longer seems specific to the "search sidebar"

**Comment by antoine-atmire** [comment]
@tdonohue all set

**Pull request merged**